// Copyright 2024 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

edition = "2023";

package optimization_guide.proto;

import "components/optimization_guide/proto/model_quality_metadata.proto";

option optimize_for = LITE_RUNTIME;
option java_package = "org.chromium.components.optimization_guide.features.proto";

option java_outer_classname = "HistoryQueryIntentProto";

// DO NOT EDIT THIS FILE DIRECTLY!
//
// This file is generated in g3 and then synced to Chrome. Instead, please refer to
// http://go/chrome-mqls-onboarding (Google-internal link), and then changes will
// be synced with Chrome automatically.

message HistoryQueryIntentLoggingData {
  HistoryQueryIntentRequest request = 1 [features = { field_presence: EXPLICIT }];

  HistoryQueryIntentResponse response = 2 [features = { field_presence: EXPLICIT }];

  ModelExecutionInfo model_execution_info = 4 [features = { field_presence: EXPLICIT }];

  reserved 3;
}

// The History Query Intent request.
message HistoryQueryIntentRequest {
  // What the user typed.
  string text = 1 [features = { field_presence: IMPLICIT }];
}

// The History Query Intent response.
message HistoryQueryIntentResponse {
  // Whether the text looks like a query.
  bool is_answer_seeking = 1 [features = { field_presence: IMPLICIT }];
}
